File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.swift
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     var val: Int
 *     var left: TreeNode?
 *     var right: TreeNode?
 *     init(_ val: Int)https://neetcode.io/$0 {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */

class Solution {
    func isValidBST(_ root: TreeNode?) -> Bool {
        func valid(_ node: TreeNode?, _ min: Int, _ max: Int) -> Bool {
            guard let node = node else{
                return true
            }
            
            if !(node.val < max && node.val > min) {
                return false
            }
            return (valid(node.left, min, node.val) && valid(node.right, node.val, max))
        
         }
     return valid(root, Int.min, Int.max)
    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:

        arr = [] 

        def inOrder(node):
            if not node: 
                return 
        
            # go left 
            inOrder(node.left)
            # append 
            arr.append(node.val)
            # go right 
            inOrder(node.right)

        inOrder(root)
        return arr
        